### PR TITLE
Constructing config handler is the analyzer classes' responsibility

### PR DESCRIPTION
**Pull request by Xazax-hun**
Another step towards https://github.com/Ericsson/codechecker/issues/1760

Also fixed a bug along the way. If clang-6.0 was available in the path but not clang, and clang-6.0 was not configured in the package config, we failed on the statistics feature check (not finding the binary).

It is hard to add a test for this scenario as there are lots of dependencies on the environment.

### CODE REVIEW

File: libcodechecker/analyze/analyzers/analyzer_clang_tidy.py [hunk 1]
```diff
@@ -14,12 +14,16 @@
...
+from libcodechecker.analyze import analyzer_env
+from libcodechecker.analyze import host_check
 from libcodechecker.analyze.analyzers import analyzer_base
-from libcodechecker.logger import get_logger
-from libcodechecker.util import get_binary_in_path
-from libcodechecker.analyze.analyzer_env import\
-    extend_analyzer_cmd_with_resource_dir
+from libcodechecker.analyze.analyzers import config_handler_clang_tidy
 from libcodechecker.analyze.analyzers import result_handler_clang_tidy
+from libcodechecker.analyze.analyzers.analyzer_clangsa import ClangSA
```
Comment (gyorb) [inline libcodechecker/analyze/analyzers/analyzer_clang_tidy.py]: Shouldn't we create a separate binary detection part for each analyzers so the tidy analyzer will not depend on the `analyzer_clangsa` modul?

File: libcodechecker/analyze/pre_analysis_manager.py [hunk 2]
```diff
@@ -49,7 +48,7 @@ def collect_statistics(action, source, config, environ, statistics_data):
     LOG.debug_analyzer(cmd)
 
     ret_code, analyzer_out, analyzer_err = \
-        analyzer_base.SourceAnalyzer.run_proc(cmdstr)
+        analyzer_base.SourceAnalyzer.run_proc(cmdstr, env=environ)
```
Comment (gyorb) [inline libcodechecker/analyze/pre_analysis_manager.py]: Maybe I just missed it but where is `environ` set to some value? It was an argument before.

File: libcodechecker/analyze/analyzers/analyzer_clang_tidy.py (hunk 1 above)
Comment (Xazax-hun) [reply libcodechecker/analyze/analyzers/analyzer_clang_tidy.py]: We definitely should. It is a workaround below in this file, that we cannot get the resource dir from clang-tidy at the moment. As soon as we come up with a way to get the resource dir without looking at the clang binary, we can get rid of this dependency. There is a fixme for that in this file.

File: libcodechecker/analyze/pre_analysis_manager.py (hunk 2 above)
Comment (Xazax-hun) [reply libcodechecker/analyze/pre_analysis_manager.py]: It is still an argument. It was just unused before. `build_stat_coll_cmd` did not use it.

**Comment by Xazax-hun** [comment]
ping :) I just resolved some merge conflicts, it would be great to have this in to avoid further rebasing

**Comment by Xazax-hun** [comment]
ping

**Comment by Xazax-hun** [comment]
I improved the commit messages and moved the fix into a separate commit. I would keep all off the commits as they are reflecting different individual steps of the refactoring.

**Comment by Xazax-hun** [comment]
ping

**Pull request merged**